processor/src/host/advice/mod.rs

```diff
@@ -269,21 +294,19 @@ impl AdviceProvider {
         self.store.get_node(root, NodeIndex::root()).is_ok()
     }
 
-    // HELPERS
-    // --------------------------------------------------------------------------------------------
+    /// Extends the [MerkleStore] with the given nodes.
+    pub fn extend_merkle_store<I>(&mut self, iter: I)
+    where
+        I: Iterator<Item = InnerNodeInfo>,
```

Comment:
Nit: this should be `IntoIterator`. Any `Iterator` implements `IntoIterator` so the latter is more general.